clique-grpc/src/server.rs

```diff
@@ -2,24 +2,27 @@ use crate::{
     proto::{self, server},
     Error,
 };
-use clique::transport::{Request, Server};
+use clique::transport::{Request, Response};
 use futures::sync::oneshot as oneshot01;
 use futures::{Future as Future01, Stream as Stream01};
-use futures03::{future, TryFutureExt};
+use futures03::TryFutureExt;
 use std::net::SocketAddr;
 use tokio_sync::{mpsc, oneshot};
 use tokio_tcp::TcpListener;
 use tower_grpc::{Code, Status};
 use tower_hyper::server::Http;
 
 pub struct GrpcServer {
-    msg_rx: Option<mpsc::Receiver<Result<Request, Error>>>,
+    msg_rx: Option<
+        mpsc::Receiver<Result<(Request, oneshot::Sender<Result<Response, clique::Error>>), Error>>,
```

Comment:
Maybe define a separate type for the inner oneshot? I feel like having a type name this long might make it difficult to understand what its doing.

Something along the lines of:

```rust
type ResponseTx = oneshot::Sender<Result<Response, clique::Error>>;

pub struct GrpcServer {
    msg_rx: Option<mpsc::Receiver<Result<(Request, ResponseTx)>, Error>>,
    target_tx: Option<oneshot01::Sender<SocketAddr>>,
}
```

Reply:
Yeah, I was gonna do that but the grpc impl is actually very incomplete for now so I plan on refactoring this down the line.